File: BE/order-service/src/main/java/com/manager/order/domain/models/entities/Order.java
package com.manager.order.domain.models.entities;

import com.manager.order.domain.models.enums.OrderStatus;
import com.manager.order.interfaces.rest.dto.response.OrderWithCreatorDTO;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@javax.persistence.Table(name = "ORDERS")
@Getter
@Setter
public class Order {

    @Id
    @Column(name = "ID")
    private String id;

    @Column(name = "TABLE_ID")
    private String tableId;

    @Column(name = "WAITER_ID")
    private String waiterId;

    @Column(name = "SERVER")
    private String server;

    @Column(name = "CREATED_AT")
    private LocalDateTime createdAt;

    @Column(name = "CREATED_BY")
    private String createdBy;

    @Enumerated(EnumType.STRING)
    @Column(name = "STATUS")
    private OrderStatus status;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<OrderItem> items = new ArrayList<>();

    @Column(name = "TOTAL_AMOUNT")
    private double totalAmount;

    public OrderWithCreatorDTO convert() {
        OrderWithCreatorDTO dto = new OrderWithCreatorDTO();
        dto.setId(id);
        dto.setTableId(tableId);
        dto.setServer(server);
        dto.setCreatedAt(createdAt);
        dto.setCreatedBy(createdBy);
        dto.setStatus(status);
        dto.setItems(items);
        dto.setTotalAmount(totalAmount);
        return dto;
    }
}

File: FE/app/src/main/java/com/food/order/data/model/Order.kt
package com.food.order.data.model

import com.google.gson.annotations.SerializedName
import java.io.Serializable

data class Order(
    @SerializedName("id") val id: String,
    @SerializedName("table_id") val tableId: String,
    @SerializedName("server") val server: String,
    @SerializedName("created_at") val createdAt: String,
    @SerializedName("created_by") val createdBy: String,
    @SerializedName("status") val status: String,
    @SerializedName("items") val items: List<OrderItem> = emptyList(),
    @SerializedName("total_amount") val totalAmount: Double,
    @SerializedName("employee_name") val employeeName: String? = null,
    @SerializedName("table_name") val tableName: String? = null
) : Serializable
